### DISCUSSION TITLE
Why removing WidthAdjustedInputStream?

### DISCUSSION THREAD

**Discussion by MohamedA95** [Xilinx/finn-hlslib#93] [Q&A] [answered]
Hi,
I have a question regarding the latest work on the Vitis-hls branch I see that you are getting rid of `WidthAdjustedInputStream`  & `WidthAdjustedOutputStream`, what is the driver behind this move? Does it lead to better performance or is it not supported any more in Vitis?

**Comment by preusser** [comment, marked as answer]
The `WidthAdjustedInputStream` worked by placing its functional kernel into its constructor. This mechanism was no longer reliable with the code transformations VitisHLS now applies in its compilation process.

**Reply by MohamedA95** [reply]
Hi Dr-ing Preußer
Thanks for the explanation.
